core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -53,7 +53,7 @@
  * retrieve the base object.
  * <p>
  * This allows us to address 8192 pages. In on-heap mode, the maximum page size is limited by the
- * maximum size of a long[] array, allowing us to address 8192 * 2^32 * 8 bytes, which is
+ * maximum size of a long[] array, allowing us to address 8192 * 2^31 * 8 bytes, which is
```

Comment:
Yes, I'm ignoring the minus one, I'll change it right away. But I didn't understand. Why not 16GB, (2^31-1)*8 = 17179869176 Bytes = 17179869176/1024/1024/1024 = 16GB. so, the amount of memory is more like 128TB(16GB * 8192/1024 = 128TB), not 140TB. I think 1024GB is 1TB, not 1000GB
(To be honest,  I was puzzled how 35TB was calculated,  and now I know that it was a miscalculation)